### ISSUE TITLE
Arithmetic regression failures on Solaris/illumos and HP-UX when compiling with gcc

### ISSUE THREAD

**Issue by McDutchie** [ksh93/ksh#175]
Attn: @lev105

The regression tests you added in https://github.com/ksh93/ksh/commit/d5a94b37224cfb9f0f3de79330ba5cf9d10bae9d are failing on at least Solaris (gcc), illumos, ~~AIX,~~ and HP-UX:

```
#### Regression-testing /usr/local/src/ksh/arch/sol11.i386/bin/ksh ####
test arith begins at 2021-02-15+03:21:25
        arith.sh[301]: 1**Inf not working
        arith.sh[304]: 1**NaN not working
        arith.sh[474]: pow(1.0,-Inf) != 1.0
        arith.sh[476]: pow(1.0,Inf) != 1.0
        arith.sh[478]: pow(1.0,NaN) != 1.0
test arith failed at 2021-02-15+03:21:26 with exit code 5 [ 208 tests 5 errors ]
```

I've tried extending [the feature test in features/float](https://github.com/ksh93/ksh/blob/5be585f45b4ba61aa61cd8951f5e9e5d9a7ba623/src/lib/libast/features/float#L1234-L1245) to account for these failures, without any effect. Clearly, I have no idea what I'm doing. In fact, I would have thought the existing feature test should cover the `pow(1.0,Inf) != 1.0` failure.

Any ideas?

**Comment by lkujaw** [comment]
Hi- sorry for not getting back to you right away, I had a pretty hectic day. This is surprising to me, because I'm pretty sure that the pow(1.0,Inf) test was already in there, I just extended it to cover the other corner cases. What was the value of _ast_pow_ieee from the test I added in FEATURE/float? Are you using the native toolchains or GCC? I've been surprised by the floating point code of the former on some occasions.

I sent in a Polarhome application, so if you don't mind the wait, hopefully I'll be able to take a look myself.

**Comment by McDutchie** [comment]
To clarify, the `pow(1.0,Inf)` test always failed on those systems, it's just that the new feature test didn't change that and the new regression tests fail equally. Sorry for any confusion.

`_ast_pow_ieee` is defined as 1 on all these systems, i.e., the feature test passes. I've tried extending the feature test with the other failure cases and it keeps passing. So somehow the problem does not seem to be in `pow`(3) itself.

I compiled ksh with gcc 5.5.0 on Solaris 10.1, gcc 7.3.0 on Solaris 11.4, gcc 8.3.0 on illumos (OmniOS 2020-08-19), and gcc 4.2.3 on HP-UX B.11.11. These all fail the arith tests identically.

I need to correct myself on one point: these tests do *not* fail on AIX (7.1), even though compilation is done with gcc 4.6.3. So I hope the polarhome application you submitted wasn't for an AIX system.

The tests also pass on Solaris 11.4 with the native toolchain -- it's only when compiling with gcc that it fails.
